### PR TITLE
feat:optimizing code

### PR DESCRIPTION
**Pull request by jinjim**
## *Tips*
- *Thanks very much for contributing to Apache DolphinScheduler.*
- *Please review https://dolphinscheduler.apache.org/en-us/community/index.html before opening a pull request.*

## What is the purpose of the pull request

optimizing AlertGroupService#grantUser by batch insert into db.

*(For example: This pull request adds checkstyle plugin.)*

## Brief change log

*(for example:)*
  - *Add maven-checkstyle-plugin to root pom.xml*

## Verify this pull request

*(Please pick either of the following options)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

AlertGroupServiceTest#testGrantUser

*(example:)*

  - *Added dolphinscheduler-dao tests for end-to-end.*
  - *Added CronUtilsTest to verify the change.*
  - *Manually verified the change by testing locally.*

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/UserAlertGroupService.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package org.apache.dolphinscheduler.api.service;
```
Comment (Jave-Chen) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/UserAlertGroupService.java]: need add license

File: dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/UserAlertGroupServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package org.apache.dolphinscheduler.api.service;
```
Comment (Jave-Chen) [inline dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/UserAlertGroupServiceTest.java]: need add license

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/UserAlertGroupService.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl;
+import org.apache.dolphinscheduler.dao.entity.UserAlertGroup;
+import org.apache.dolphinscheduler.dao.mapper.UserAlertGroupMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * Created on 2020/3/19.
+ *
+ * @author flowkr90@gmail.com
```
Comment (Jave-Chen) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/UserAlertGroupService.java]: author needs removed

File: dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/UserAlertGroupServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * Created on 2020/3/19.
+ *
+ * @author flowkr90gmail.com
```
Comment (Jave-Chen) [inline dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/UserAlertGroupServiceTest.java]: author needs removed

**Comment by dailidong** [comment]
good beginning

**Comment by dailidong** [comment]
suggestion: 
1、create an issue for this PR , tell this PR used for what, one issue describe one thing
2、one PR for one issue, so one PR do one thing 
3、PR title should describe the detail thing , use 'feat:optimizing code' is not good

**Comment by dailidong** [comment]
> suggestion:
> 1、create an issue for this PR , tell this PR used for what, one issue describe one thing
> 2、one PR for one issue, so one PR do one thing
> 3、PR title should describe the detail thing , use 'feat:optimizing code' is not good

because this is your first PR , just pay attention next time

**Comment by jinjim** [comment]
> > suggestion:
> > 1、create an issue for this PR , tell this PR used for what, one issue describe one thing
> > 2、one PR for one issue, so one PR do one thing
> > 3、PR title should describe the detail thing , use 'feat:optimizing code' is not good
> 
> because this is your first PR , just pay attention next time

ok!

**Comment by dailidong** [comment]
hi，you can add my wechat(510570367) or mail(dailidong66@gmail.com), if you'd like to enter  developer wechat group, please tell me ,when added,  tell me your github id

thx

**Pull request merged**